cirq/google/optimizers/convert_to_sqrt_iswap.py

```diff
@@ -0,0 +1,290 @@
+        """
+        if isinstance(op, cirq.GlobalPhaseOperation):
+            return []
+
+        gate = cast(cirq.GateOperation, op).gate
+
+        if len(op.qubits) != 2:
+            return NotImplemented
+
+        if isinstance(gate, cirq.CZPowGate):
+            turns = cast(cirq.CZPowGate, gate).exponent
+            if isinstance(turns, sympy.Basic):
+                return cphase_symbols_to_sqrt_iswap(op.qubits[0], op.qubits[1],
+                                                    turns)
+            else:
+                return cphase_to_sqrt_iswap(op.qubits[0], op.qubits[1], turns)
+        if isinstance(gate, cirq.SwapPowGate):
+            return swap_to_sqrt_iswap(op.qubits[0], op.qubits[1],
+                                      cast(cirq.SwapPowGate, gate).exponent)
+        if isinstance(gate, cirq.ISwapPowGate):
+            return iswap_to_sqrt_iswap(op.qubits[0], op.qubits[1],
+                                       cast(cirq.ISwapPowGate, gate).exponent)
+        if isinstance(gate, cirq.FSimGate):
+            return fsim_gate(op.qubits[0], op.qubits[1],
+                             cast(cirq.FSimGate, gate).theta,
+                             cast(cirq.FSimGate, gate).phi)
+
+        return NotImplemented
+
+    def convert(self, op: cirq.Operation) -> List[cirq.Operation]:
+
+        def on_stuck_raise(bad):
+            return TypeError("Don't know how to work with {!r}. "
+                             "It isn't a native fishfood operation, "
+                             "a 1 or 2 qubit gate with a known unitary, "
+                             "or composite.".format(bad))
+
+        a = cirq.decompose(
+            op,
+            keep=is_sqrt_iswap_fishfood_compatible,
+            intercepting_decomposer=self._convert_one,
+            on_stuck_raise=None if self.ignore_failures else on_stuck_raise)
+        return a
+
+    def optimization_at(self, circuit, index, op):
+        converted = self.convert(op)
+        if len(converted) == 1 and converted[0] is op:
+            return None
+
+        return cirq.PointOptimizationSummary(clear_span=1,
+                                             new_operations=converted,
+                                             clear_qubits=op.qubits)
+
+
+def is_sqrt_iswap_fishfood_compatible(op: cirq.Operation) -> bool:
+    """Check if the given operation is compatible with the SQRT_ISWAP_FISHFOOD
+    gate set.
+
+    Args:
+        op: Input operation.
+
+    Returns:
+        True if the operation is native to the gate set, false otherwise.
+    """
+    return (isinstance(op, cirq.GateOperation) and
+            (is_basic_gate(op.gate) or is_sqrt_iswap(op.gate)))
+
+
+def is_sqrt_iswap(gate: cirq.Gate) -> bool:
+    """Checks if this is a ± sqrt(iSWAP) gate specified using either
+    ISwapPowGate or with the equivalent FSimGate.
+    """
+    if (isinstance(gate, cirq.FSimGate) and
+            not isinstance(gate.theta, sympy.Symbol) and
```

Comment:
`gate.theta` could be a more complicated expression than just a symbol. I think this should check `sympy.Basic` instead, though it's probably safer to use `cirq.is_parametrized`. Same thing for the `ISwapPowGate` case below.